spyder/api/plugins.py

```diff
@@ -152,10 +152,11 @@ def register_widget_shortcuts(self, widget):
     def visibility_changed(self, enable):
         """Dock widget visibility has changed."""
```

Comment:
I think this could be made simpler if we add at this point:

```
if self.dockwidget is None:
    return
```

This way we can avoid all instructions below that depend on the existence of `dockwidget`.